api/src/main/resources/applicationContext-service.xml

```diff
@@ -365,175 +233,104 @@
-		<property name="transactionManager" ref="transactionManager"/>
-		<property name="target" ref="serializationServiceTarget"/>
-		<property name="preInterceptors" ref="serviceInterceptors"/>
-		<property name="transactionAttributeSource">
-			<bean class="org.springframework.transaction.annotation.AnnotationTransactionAttributeSource"/>
-		</property>
-	</bean>
-	<bean id="hL7Service" class="org.springframework.transaction.interceptor.TransactionProxyFactoryBean">
-		<property name="transactionManager" ref="transactionManager"/>
-		<property name="target" ref="hL7ServiceTarget"/>
-		<property name="preInterceptors" ref="serviceInterceptors"/>
-		<property name="transactionAttributeSource" ref="transactionAttributeSource"/>
-	</bean>
-	<bean id="messageSourceService" class="org.springframework.transaction.interceptor.TransactionProxyFactoryBean">
-		<property name="transactionManager" ref="transactionManager"/>
-		<property name="target" ref="messageSourceServiceTarget"/>
-		<property name="preInterceptors" ref="serviceInterceptors"/>
-		<property name="transactionAttributeSource" ref="transactionAttributeSource"/>
-	</bean>
-	<bean id="visitService" class="org.springframework.transaction.interceptor.TransactionProxyFactoryBean">
-		<property name="transactionManager" ref="transactionManager"/>
-		<property name="target" ref="visitServiceTarget"/>
-		<property name="preInterceptors" ref="serviceInterceptors"/>
-		<property name="transactionAttributeSource" ref="transactionAttributeSource"/>
-	</bean>
-	<bean id="providerService" class="org.springframework.transaction.interceptor.TransactionProxyFactoryBean">
-		<property name="transactionManager" ref="transactionManager"/>
-		<property name="target" ref="providerServiceTarget"/>
-		<property name="preInterceptors" ref="serviceInterceptors"/>
-		<property name="transactionAttributeSource" ref="transactionAttributeSource"/>
-	</bean>
+	<bean id="hL7Parser" class="ca.uhn.hl7v2.parser.GenericParser"/>
 
+	<bean id="hL7Router" class="ca.uhn.hl7v2.app.MessageTypeRouter"/>
+	
+	<util:map id="hL7Handlers" key-type="java.lang.String" value-type="ca.uhn.hl7v2.app.Application">
+		<entry key="ORU_R01">
+			<bean class="org.openmrs.hl7.handler.ORUR01Handler"/>
+		</entry>
+		<entry key="ADT_A28">
+			<bean class="org.openmrs.hl7.handler.ADTA28Handler"/>
+		</entry>
+	</util:map>
 
 	<!--  **************************  SERVICE INTERCEPTORS  *************************  -->
 	<!-- AOP before advice that authorizes users according to annotations -->
 	<bean id="authorizationInterceptor" class="org.openmrs.aop.AuthorizationAdvice"/>
+	
+	<bean id="authorizationAdvisor" class="org.springframework.aop.support.DefaultPointcutAdvisor">
+		<property name="pointcut">
+			<bean class="org.springframework.aop.aspectj.AspectJExpressionPointcut">
+				<property name="expression" value="this(org.openmrs.api.OpenmrsService)" />
+			</bean>
+		</property>
+		<property name="advice" ref="authorizationInterceptor" />
```

Comment:
Let's get this into:
```
@Configuration 
public class AOPConfig {
	@Autowired
	private LoggingAdvice loggingAdvice;

	@Bean
	public Advisor loggingAdvisor() {
		return new StaticMethodMatcherPointcutAdvisor(loggingAdvice) {
			@Override
			public boolean matches(Method method, Class<?> targetClass) {
				return targetClass.isInstance(OpenmrsService.class);
			}
		};
	}
}
```
and move away from xml even further :)

Reply:
I was thinking I could do that also instead of xml, just wanted to know if it was okay